R/translate-sql-string.R

```diff
@@ -0,0 +1,9 @@
+# R prefers to specify start / stop or start / end
+# databases usually specify start / length
+# https://www.postgresql.org/docs/current/functions-string.html
+sql_substr <- function(x, start, stop) {
```

Comment:
To be consistent with the other helper functions, this should be a function factory that takes the name of the SQL function as an input. Look at `sql_paste()` (but `sql_substr()` should be a bit simpler), and then let me know if you need more help.